File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.py
class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        costs.sort()
        count=0
        
        n=len(costs)
        summ=0
        for i in range(n):
            summ+=costs[i]
            if summ>coins:
                break
            count+=1 
        return count          

        

File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.java
class Solution {
    public int maxIceCream(int[] costs, int coins) {
        Arrays.sort(costs);
        int count=0;
        int n= costs.length;
        int sum=0;
        for (int i=0;i<n;i++){
            sum+=costs[i];
            if (sum>coins){
                break;
            }
            count+=1;
        }
        return count;
        
    }
}